lib/trino-memory-context/src/main/java/io/trino/memory/context/MemoryAllocationValidator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.memory.context;
+
+import java.util.function.BiFunction;
+
+public interface MemoryAllocationValidator
+{
+    MemoryAllocationValidator NO_MEMORY_VALIDATION = new MemoryAllocationValidator() {
+        @Override
+        public void reserveMemory(String allocationTag, long delta) {}
+
+        @Override
+        public boolean tryReserveMemory(String allocationTag, long delta)
+        {
+            return true;
+        }
+    };
+
+    static MemoryAllocationValidator newAllocationValidator(
```

Comment:
These two methods seem to be unused

Reply:
Yeah - remainder before I structured code differently. Removed.